File: redsheet-portal/src/app/database-models/country.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { State, CultureCountry, Culture,  } from './'
export class Country {
    constructor(country?: Country )  {
        let properties = [ "active", "countryCode", "countryId", "countryName", "currencyFormat", "dateFormat", "dateFormatSeperator", "decimalSeperator", "defaultLanguageId", "phoneFormat", "postalCodeFormat", "states", "cultureCountries", "cultures",];
        for (let property of properties)
            if (country && country[property])
                this[property] = country[property];
    }
 
	active : boolean = false ;
 
    @required()
    @maxLength(2)
	countryCode : string =   undefined;
 
	countryId : number =   0 ;
 
    @required()
    @maxLength(50)
	countryName : string =   undefined;
 
    @maxLength(20)
	currencyFormat : string =   undefined;
 
    @maxLength(20)
	dateFormat : string =   undefined;
 
    @maxLength(1)
	dateFormatSeperator : string =   undefined;
 
    @maxLength(10)
	decimalSeperator : string =   undefined;
 
    @range(1,2147483647)
	defaultLanguageId : number =   undefined;
 
    @maxLength(20)
	phoneFormat : string =   undefined;
 
    @maxLength(20)
	postalCodeFormat : string =   undefined;
	@nested(State)
	states: State[];

	@nested(CultureCountry)
	cultureCountries: CultureCountry[];

	@nested(Culture)
	cultures: Culture[];



}

File: RedSheet.DbEntities/Models/Country.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("Countries", Schema = "dbo")]
	[Serializable]
	public partial class Country  
	{
 
	[Required]
	public bool Active { get; set; } 
 
	[Required] 
	[MaxLength(2)]
	public string CountryCode { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int CountryId { get; set; } 
 
	[Required] 
	[MaxLength(50)]
	public string CountryName { get; set; } 
 
	[MaxLength(20)]
	public string CurrencyFormat { get; set; } 
 
	[MaxLength(20)]
	public string DateFormat { get; set; } 
 
	[MaxLength(1)]
	public string DateFormatSeperator { get; set; } 
 
	[MaxLength(10)]
	public string DecimalSeperator { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int DefaultLanguageId { get; set; } 
 
	[MaxLength(20)]
	public string PhoneFormat { get; set; } 
 
	[MaxLength(20)]
	public string PostalCodeFormat { get; set; } 
	[InverseProperty("Country")]
	public ICollection<State> States { get; set; } 

	[InverseProperty("Country")]
	public ICollection<CultureCountry> CultureCountries { get; set; } 

	[InverseProperty("Country")]
	public ICollection<Culture> Cultures { get; set; } 


	public Country()
	{
	this.States = new HashSet<State>();

	this.CultureCountries = new HashSet<CultureCountry>();

	this.Cultures = new HashSet<Culture>();

    }

    }
}

